File: MATTHEW/leetcode/AIcoding/5.Qwen-3/java_solutions/1576-replace-all-s-to-avoid-consecutive-repeating-characters.java
public class Solution {
    public String modifyString(String s) {
        char[] chars = s.toCharArray();
        for (int i = 0; i < chars.length; i++) {
            if (chars[i] == '?') {
                for (char c = 'a'; c <= 'z'; c++) {
                    boolean valid = true;
                    if (i > 0 && chars[i - 1] == c) {
                        valid = false;
                    }
                    if (i < chars.length - 1 && chars[i + 1] == c) {
                        valid = false;
                    }
                    if (valid) {
                        chars[i] = c;
                        break;
                    }
                }
            }
        }
        return new String(chars);
    }
}

File: leetcode/AIcoding/4.GPT-4o/go_solutions/1576-replace-all-s-to-avoid-consecutive-repeating-characters.go
func modifyString(s string) string {
    n := len(s)
    chars := []byte(s)

    for i := 0; i < n; i++ {
        if chars[i] == '?' {
            for c := 'a'; c <= 'z'; c++ {
                if (i == 0 || chars[i-1] != byte(c)) && (i == n-1 || chars[i+1] != byte(c)) {
                    chars[i] = byte(c)
                    break
                }
            }
        }
    }

    return string(chars)
}